src/controllers/library.py

```diff
@@ -34,55 +37,52 @@ def __init__(self, **data):
         """
         Add new library items to the library system.
 
         Args:
             new_library_items (List[LibraryItem]): List of library items to add.
         """
         for new_library_item in new_library_items:
-            if new_library_item.isbn in self.library_items.keys():
+            if LibraryItemDocument.objects(isbn=new_library_item.isbn).count() > 0:
                 raise ItemAlreadyExistsError(new_library_item.isbn)
-            self.library_items[new_library_item.isbn] = new_library_item
             new_library_item_document = item_pydantic_to_mongoengine(new_library_item)
             self.controllers_manager.library_item_repo.insert_document(new_library_item_document)
             logging.info(
                 f"The library item {new_library_item.title} with ISBN {new_library_item.isbn} "
                 f"was added to '{self.name}' library"
             )
 
-    def add_new_patron_to_the_library(self, patrons_to_add: List[Patron]):
+    def add_new_patrons(self, patrons_to_add: List[Patron]):
         """
         Add new patrons to the library system.
 
         Args:
             patrons_to_add (List[Patron]): List of patrons to add.
         """
         for patron in patrons_to_add:
-            if patron.patron_id in self.patrons.keys():
+            if PatronDocument.objects(patron_id=patron.patron_id).count() > 0:
                 raise PatronAlreadyExistsError(patron.patron_id)
-            self.patrons[patron.patron_id] = patron
             patron_document = patron_pydantic_to_mongoengine(patron)
             self.controllers_manager.patron_repo.insert_document(patron_document)
             logging.info(f"The patron {patron.name} with ID {patron.patron_id} was added to the library")
 
-    def remove_patrons_from_the_library(self, patrons_to_remove: List[Patron]):
+    def remove_patron(self, patron_id: str = Field(..., pattern=r'^\d{9}$')):
         """
         Remove patrons from the library system.
 
         Args:
             patrons_to_remove (List[Patron]): List of patrons to remove.
+            :param patron_id:
         """
 
-        for patron in patrons_to_remove:
-            if patron.patron_id not in self.patrons:
-                raise PatronNotFoundError(patron.patron_id)
-            del self.patrons[patron.patron_id]
-            self.controllers_manager.patron_repo.delete_document({'patron_id': patron.patron_id})
-            logging.info(f"The patron {patron.patron_id} was removed from the library")
+        if PatronDocument.objects(patron_id=patron_id).count() == 0:
+            raise PatronNotFoundError(patron_id)
+        self.controllers_manager.patron_repo.delete_document({'patron_id': patron_id})
+        logging.info(f"The patron {patron_id} was removed from the library")
 
-    def search_library_items(self, library_item_title=None, library_item_isbn=None):
+    def search_items(self, library_item_title=None, library_item_isbn=None):
```

Comment:
When searching, it makes more sense to input a string that coult be an isbn or a title.
Maybe refactor this to be agnostic to what type of input it recieves?